lib/html_processor/amp_img.rb

```diff
@@ -0,0 +1,62 @@
+require_dependency 'html_processor/base'
+
+module HTMLProcessor
+  class AmpImg < Base
+    PERMITTED_AMP_ATTRIBUTES = %w[
+      src
+      srcset
+      sizes
+      alt
+      attribution
+      height
+      width
+    ].freeze
+
+    def process(*xpaths)
+      doc.xpath(*xpaths).each do |original_img|
+        if original_img.attributes.include? 'src'
+          container = create_container
+          amp_img = create_amp_img(original_img)
+          container.add_child(amp_img)
+          original_img.replace container
```

Comment:
Methods with generic names like 'process' and/or more than 4 lines often indicate that it is violating the single responsibility principle. Perhaps this could be refactored, e.g.
```
if original_img.attributes.include? 'src'
  replace_original_img
else
  original_img.remove
end

def replace_original_img(original_img)
  container = create_container
  amp_img = create_amp_img(original_img)
  container.add_child(amp_img)
  original_img.replace container
end